hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -286,13 +293,29 @@ public Hedera(
         try {
             // And the factory for the MerkleStateRoot class id must be our constructor
             constructableRegistry.registerConstructable(new ClassConstructorPair(
-                    MerkleStateRoot.class, () -> new MerkleStateRoot(new MerkleStateLifecyclesImpl(this))));
+                    MerkleStateRoot.class,
+                    () -> new MerkleStateRoot(new MerkleStateLifecyclesImpl(this), softwareVersionSupplier)));
         } catch (final ConstructableRegistryException e) {
             logger.error("Failed to register " + MerkleStateRoot.class + " factory with ConstructableRegistry", e);
             throw new IllegalStateException(e);
         }
     }
 
+    private void registerPlatformStateSchema() {
+        servicesRegistry.register(new Service() {
```

Comment:
Shouldn't it be extracted to a separate class?

Reply:
I don't see a reason why it should be. It's not going be used anywhere else and it plays a role of connecting tissue.

Reply:
Just for consistency. For example, a person may need to find all services (or all schemas from all services). To me, it's worth extracting to an outer class, even if that class is just 20 lines long. It may also be a good place for JavaDoc